Error while compiling versions 0.8.0 and above on Windows

## Environment details
Windows 10, nodejs version 8.9.3

- OS: Windows 
- OS version: 10
- node-pty version: 0.8.0 and above

## Issue description

npm install fails on Windows with a number of C++ errors. I had to downgrade to version 0.7.8 to make it work. Below is the output of npm install:

$ npm install

> node-pty@0.8.1 install node_modules\node-pty
> node scripts/install.js

```

node_modules\node-pty>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  conpty.cc
  path_util.cc
  win_delay_load_hook.cc
node_modules\node-pty\src\win\conpty.cc(310): error C2065: 'STARTUPINFOEXW': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(310): error C2146: syntax error: missing ';' before identifier 'siEx' [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(310): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(311): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(311): error C2065: 'STARTUPINFOEXW': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(313): error C3861: 'InitializeProcThreadAttributeList': identifier not found [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(315): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(317): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(317): error C3861: 'InitializeProcThreadAttributeList': identifier not found [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(321): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(323): error C3861: 'ProcThreadAttributeValue': identifier not found [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(321): error C3861: 'UpdateProcThreadAttribute': identifier not found [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(342): error C2065: 'siEx': undeclared identifier [node_modules\node-pty\build\conpty.vcxproj]
node_modules\node-pty\src\win\conpty.cc(333): error C2660: 'CreateProcessW': function does not take 9 arguments [node_modules\node-pty\build\conpty.vcxproj]
  c:\program files (x86)\windows kits\10\include\10.0.17134.0\um\processthreadsapi.h(389): note: see declaration of 'CreateProcessW' (compiling source file ..\src\win\conpty.cc)
  conpty_console_list.cc
  win_delay_load_hook.cc
     Creating library node_modules\node-pty\build\Release\conpty_console_list.lib and object node_modules\node-pty\build\Release\conpty_console_list.exp
  Generating code
  Finished generating code
  conpty_console_list.vcxproj -> node_modules\node-pty\build\Release\\conpty_console_list.node
  AgentLocation.cc
  winpty.cc
  BackgroundDesktop.cc
  Buffer.cc
  DebugClient.cc
  GenRandom.cc
  OwnedHandle.cc
  StringUtil.cc
  WindowsSecurity.cc
  WindowsVersion.cc
  WinptyAssert.cc
  WinptyException.cc
  WinptyVersion.cc
  win_delay_load_hook.cc
     Creating library node_modules\node-pty\build\Release\winpty.lib and object node_modules\node-pty\build\Release\winpty.exp
  Generating code
  Finished generating code
  winpty.vcxproj -> node_modules\node-pty\build\Release\\winpty.dll
  Agent.cc
  AgentCreateDesktop.cc
  ConsoleFont.cc
  ConsoleInput.cc
  ConsoleInputReencoding.cc
  ConsoleLine.cc
  DebugShowInput.cc
  DefaultInputMap.cc
  EventLoop.cc
node_modules\node-pty\deps\winpty\src\agent\defaultinputmap.cc(291): warning C4244: 'argument': conversion from '__int64' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
node_modules\node-pty\deps\winpty\src\agent\consoleinput.cc(121): warning C4244: 'return': conversion from '__int64' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
node_modules\node-pty\deps\winpty\src\agent\consoleinput.cc(137): warning C4244: 'return': conversion from '__int64' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
node_modules\node-pty\deps\winpty\src\agent\consoleinput.cc(155): warning C4244: 'return': conversion from '__int64' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
node_modules\node-pty\deps\winpty\src\agent\consoleinput.cc(172): warning C4244: 'return': conversion from '__int64' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
  InputMap.cc
  LargeConsoleRead.cc
  NamedPipe.cc
  Scraper.cc
  Terminal.cc
  Win32Console.cc
  Win32ConsoleBuffer.cc
  main.cc
node_modules\node-pty\deps\winpty\src\agent\scraper.cc(560): warning C4244: 'argument': conversion from 'int64_t' to 'const _Ty', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
          with
          [
              _Ty=int
          ]
node_modules\node-pty\deps\winpty\src\agent\scraper.cc(623): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
  BackgroundDesktop.cc
  Buffer.cc
  DebugClient.cc
  GenRandom.cc
  OwnedHandle.cc
  StringUtil.cc
  WindowsSecurity.cc
  WindowsVersion.cc
  WinptyAssert.cc
  WinptyException.cc
  WinptyVersion.cc
  win_delay_load_hook.cc
  Generating code
node_modules\node-pty\deps\winpty\src\agent\agent.cc(231): warning C4722: 'Agent::~Agent': destructor never returns, potential memory leak [node_modules\node-pty\build\deps\winpty\src\winpty-agent.vcxproj]
  Finished generating code
  winpty-agent.vcxproj -> node_modules\node-pty\build\Release\\winpty-agent.exe
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd node_modules\node-pty
gyp ERR! node -v v8.9.3
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fs-xattr@0.1.17 (node_modules\fs-xattr):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fs-xattr@0.1.17: wanted {"os":"!win32","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: appdmg@0.5.2 (node_modules\appdmg):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for appdmg@0.5.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: macos-alias@0.2.11 (node_modules\macos-alias):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for macos-alias@0.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-pty@0.8.1 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-pty@0.8.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


```

--- a/src/win/conpty.cc
+++ b/src/win/conpty.cc
@@ -8,6 +8,12 @@
  *   with pseudo-terminal file descriptors.
  */
 
+// node versions lower than 10 define this as 0x502 which disables many of the definitions needed to compile
+#include <node_version.h>
+#if NODE_MODULE_VERSION <= 57
+  #define _WIN32_WINNT 0x600
+#endif
+
 #include <iostream>
 #include <nan.h>
 #include <Shlwapi.h> // PathCombine, PathIsRelative

